benchmarks/dbscan/dbscan_timpl.hpp

```diff
@@ -11,20 +11,53 @@
 
 #include <ArborX_DBSCAN.hpp>
 #include <ArborX_DBSCANVerification.hpp>
-#include <ArborX_DetailsHeap.hpp>
-#include <ArborX_DetailsOperatorFunctionObjects.hpp> // Less
 #include <ArborX_MinimumSpanningTree.hpp>
-#include <ArborX_Version.hpp>
 
 #include <Kokkos_Core.hpp>
 
-#include <boost/program_options.hpp>
-
 #include <cstdlib>
 #include <fstream>
 
-std::vector<ArborX::Point> loadData(std::string const &filename,
-                                    bool binary = true, int max_num_points = -1)
+#include "dbscan.hpp"
+
+using ArborX::ExperimentalHyperGeometry::Point;
+
+template <int DIM>
+std::vector<Point<DIM>> sampleData(std::vector<Point<DIM>> const &data,
+                                   int num_samples)
+{
+  std::vector<Point<DIM>> sampled_data(num_samples);
+
+  std::srand(1337);
+
+  // Knuth algorithm
+  auto const N = (int)data.size();
+  auto const M = num_samples;
+  for (int in = 0, im = 0; in < N && im < M; ++in)
+  {
+    int rn = N - in;
+    int rm = M - im;
+    if (std::rand() % rn < rm)
+      sampled_data[im++] = data[in];
+  }
+  return sampled_data;
+}
```

Comment:
Not having this defined next to `getDataDimensions` hurts readability

Reply:
Not sure what you are talking about. This function is completely independent from anything else.

Reply:
Both function implementation are looking into the same input data and you move them away from each other

Reply:
But functionally they have nothing to do with each other. Sampling data can be done with any input, independent of how you construct it. For example, with Gan-Tao generator you would generate the data without reading it in, and would still be able to sample it. So for me I see zero reason to keep them in the same place.

Reply:
I thought I was commenting about `loadData`.  Where did that function go?

Reply:
It seemed to me that you were commenting about `getDataDimensions`. Nevertheless, I stand by my point that sampling has nothing to do with how the data is constructed, and thus is unnecessary to be kept together. However, I'll move the `sampleData` after `loadData` if it makes you happier.